File: Maximum Unique Subarray Sum After Deletion.go
func maxSum(nums []int) int {
    seen := make(map[int]interface{})
    summ := 0
    maxx := nums[0]

    for _, num := range nums {
        if num > 0 {
            if _, ok := seen[num]; !ok {
                summ += num
                seen[num] = nil
            }
        }
        maxx = max(maxx, num)
    }

    if len(seen) == 0 {
        return maxx
    }
    return summ
}

File: Maximum Unique Subarray Sum After Deletion.py
class Solution {
public:
    int maxSum(vector<int>& nums) {
        int maxx = nums[0];
        unordered_set<int> seen;
        int summ = 0;
        for (int num: nums) {
            if (num > 0) {
                if (!seen.contains(num)) {
                    summ += num;
                    seen.insert(num);
                }
            }
            maxx = max(maxx, num);
        }
        if (seen.size() == 0) return maxx;
        return summ;
    }
};